File: ios/Source/Extensions/Credentials+WalletLink.swift
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

import CBHTTP

extension Credentials {
    init(sessionId: String, secret: String) {
        self.init(username: sessionId, password: Credentials.createSessionKey(sessionId: sessionId, secret: secret))
    }

    private static func createSessionKey(sessionId: String, secret: String) -> String {
        return "\(sessionId), \(secret) WalletLink".sha256()
    }
}

File: android/walletlink/src/main/java/com/coinbase/walletlink/extensions/Credentials+WalletLink.kt
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

package com.coinbase.walletlink.extensions

import com.coinbase.wallet.crypto.extensions.sha256
import com.coinbase.wallet.http.models.Credentials

fun Credentials.Companion.create(sessionId: String, secret: String): Credentials {
    val password = "$sessionId, $secret WalletLink".sha256()
    return Credentials(username = sessionId, password = password)
}
